### PR TITLE
consensus: add nil check to gossip routine

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: internal/consensus/reactor.go [hunk 1]
```diff
@@ -836,10 +836,20 @@ func (r *Reactor) queryMaj23Routine(ctx context.Context, ps *PeerState, stateCh
 			return
 		}
 
-		rs := r.getRoundState()
-		prs := ps.GetRoundState()
 		// TODO create more reliable coppies of these
```
Comment (williambanfield) [inline internal/consensus/reactor.go:839]:
```suggestion
		// TODO create more reliable copies of these
```

**Review by williambanfield — COMMENTED** [review]
Are we seeing crashes as a result of this?

**Review by creachadair — APPROVED** [review]
This seems ok, but could this get stuck forever waiting? Previously we'd have panicked, which would ensure at least the operator knows what happened.

I still think it's worthwhile to avoid a panic, I only wonder if we ought to cap how long we'll wait before we make some alert noise. At steady state I imagine we _do_ want to wait indefinitely, but the startup case is more interesting. 

Probably a larger issue.

**Comment by tychoish** [comment]
We're seeing panics for this in e2e tests which is causing these tests to stall (because we wait I believe it only happens when a peer connects to a node before the first round starts, and is only really a problem because there's shared (but mutexed) state in the reactor.

**Comment by tychoish** [comment]
> I still think it's worthwhile to avoid a panic, I only wonder if we ought to cap how long we'll wait before we make some alert noise. At steady state I imagine we _do_ want to wait indefinitely, but the startup case is more interesting. 

This is just in one of the per-peer threads inside of the consensus reactor, so the worst thing that happens is that we (eventually) timeout the peer (or vice-versa) though I think eventually the first round will start and then this state gets populated.

**Comment by williambanfield** [comment]
I'm not certain I see how Votes would ever be `nil`. We populate `RoundState` from the consensus reactor on construction of this reactor. Additionally, we set the `Votes` field to a non-nil struct on construction on the construction of the `State` object, so I'm not sure I see how this would be the cause yet:
https://github.com/tendermint/tendermint/blob/master/internal/consensus/state.go#L806

**Comment by tychoish** [comment]
And yet: https://github.com/tendermint/tendermint/runs/5889655157?check_suite_focus=true#step:6:1489

**Pull request merged**